### PR TITLE
Factor out @lit/localize-tools package

### PR DESCRIPTION
**Pull request by aomarks**
Previously, `@lit/localize` included both the client library and the CLI. Now, it only includes the client library, and the CLI is distributed in this new `@lit/localize-tools` package.

**Note** The `id-generation.ts` and `fnv1a64.ts` files are not actually new, the diff view is just not picking up the move correctly because a symlink has become a proper file.

Fixes https://github.com/Polymer/lit-html/issues/1429

### CODE REVIEW

File: packages/localize/tsconfig.json [hunk 1]
```diff
@@ -1,20 +1,22 @@
 {
   "compilerOptions": {
-    "target": "es2020",
-    "module": "commonjs",
+    "target": "es2018",
```
Comment (justinfagnani) [inline packages/localize/tsconfig.json:3]: why the target change?

**Review by justinfagnani — APPROVED** [review]

**Review by rictic — APPROVED** [review]

File: packages/localize-tools/CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## Unreleased
+
+- Initial release of `@lit/localize-tools` package. This new package provides
+  the `lit-localize` binary, while `@lit/localize` continues to provide the
```
[the remark this answers is not in the document]
Comment (justinfagnani) [reply packages/localize-tools/CHANGELOG.md:11]: It's not just Google, npm has a "bin" field in package.json: https://docs.npmjs.com/cli/v7/configuring-npm/package-json#bin

File: packages/localize/tsconfig.json (hunk 1 above)
Comment (aomarks) [reply packages/localize/tsconfig.json:3]: Not really changed, this tsconfig used to be for the CLI, but now it's for the client. It's actually the same as what was `tsconfig.client.json`

File: packages/localize-tools/CHANGELOG.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (aomarks) [reply packages/localize-tools/CHANGELOG.md:11]: Changed to CLI in this comment.

**Pull request merged**